utils/zerovec/src/map/serde.rs

```diff
@@ -6,11 +6,171 @@ use super::{MutableZeroVecLike, ZeroMap, ZeroMapBorrowed, ZeroMapKV, ZeroVecLike
 use core::fmt;
 use core::marker::PhantomData;
 use serde::de::{self, Deserialize, Deserializer, MapAccess, SeqAccess, Visitor};
-#[cfg(feature = "serde_serialize")]
+#[cfg(feature = "serde")]
 use serde::ser::{Serialize, SerializeMap, SerializeSeq, Serializer};
 
-/// This impl can be made available by enabling the optional `serde_serialize` feature of the `zerovec` crate
-#[cfg(feature = "serde_serialize")]
+#[cfg(feature = "serde")]
+fn is_num_or_string<T: Serialize + ?Sized>(k: &T) -> bool {
```

Comment:
Question: Why did you name it `is_num_or_string` here but `is_valid_json_key` in the other crate?

Reply:
So we can discuss :). This code is actually used for any human-readable serialization, but the logic is JSON-specific, so I'm not sure what to name this.